Add a simple command-line program to encrypt and decrypt

Crosse opened the pull request:
The README states that you can run `go test` with some environment variables set, and while that works, it could be made a little better. The command-line program can take input from stdin or a file, and will produce its output to either stdout or another file.

Example usage:

    $ echo "hello, world" | ./rot13 -encode
    uryy|9-�|yq

    $ echo "hello, world" | ./rot13 -encode | ./rot13
    hello, world

Crosse committed:
Add a simple command-line program to encrypt and decrypt

The README states that you can run `go test` with some environment variables
set, and while that works, it could be made a little better. The
command-line program can take input from stdin or a file, and will produce
its output to either stdout or another file.

Example usage:

    $ echo "hello, world" | ./rot13 -encode
    uryy|9-�|yq

    $ echo "hello, world" | ./rot13 -encode | ./rot13
    hello, world

soltzen reviewed (changes requested):
LGTM. I took a look at this repo before. I kinda still like the test though :D

That being said, would be nice to modify that part in the README to keep things easy for newcomers to find.

soltzen commented on cmd/rot13.go:
```diff
@@ -0,0 +1,61 @@
+)
+
+func die(format string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, format, args...)
```
This is missing an `os.Exit(1)`, right?

Crosse replied:
Yep, not sure where it ran off to. Added.

soltzen commented on cmd/rot13.go:
```diff
@@ -0,0 +1,61 @@
+	flag.Parse()
+
+	var in io.Reader
+	if *inFile == "-" || *inFile == "" {
```
Would be nice to document this feature in the usage string.
Also, would be nice to have a usage string (i.e., `./rot13 -h`)

Crosse replied:
I added that to the usage string. Passing `-h` does already work, though! (See updated README in a minute for an example.)

oxtoacart commented:
I don't have anything to add beyond @soltzen 's comments.

Crosse committed:
Update README

This adds usage information for the new CLI tool, and also addresses
https://github.com/getlantern/rot13/pull/1 by adding a "Caveats" section at
the top of the document.

Closes #1.

Crosse commented:
I think I've addressed the issues, @soltzen. Let me know what you think!

oxtoacart commented on README.md:
```diff
@@ -1,9 +1,60 @@
-For a standalone decode/encode, supply the input/output file path as env variables and run:
+## Caveat
+
+The name of this package is something of a misnomer: it does not implement
```
Ha, I hadn't realized that our "rot13" wasn't even rot13!

Crosse replied:
Yeah, there's an open PR (#1) pointing it out, and I also got tripped up by it when I was trying to decode something of ours, so I figured it would be good to mention!

soltzen commented:
LGTM. Go with God

Crosse commented:
I'm going to squash these commits and then merge.
